server/src/main/java/org/elasticsearch/index/analysis/AnalyzerComponents.java

```diff
@@ -44,6 +44,12 @@ static AnalyzerComponents createComponents(
         final Map<String, CharFilterFactory> charFilters,
         final Map<String, TokenFilterFactory> tokenFilters
     ) {
+        for (String key : analyzerSettings.keySet()) {
+            switch (key) {
+                case "tokenizer", "char_filter", "filter", "type", "position_increment_gap" -> {}
+                default -> throw new IllegalArgumentException("Custom Analyzer not support [" + key + "] now");
```

Comment:
could you rephrase the error to `"Custom analyzer [" + name + "] does not support [" + key + "]"` ?